packages/graphql-typescript-definitions/src/index.ts

```diff
@@ -240,21 +312,108 @@ export class Builder extends EventEmitter {
     }
   }
 
-  private async updateDocumentForFile(file: string) {
-    const contents = await readFile(file, 'utf8');
-    if (contents.trim().length === 0) {
-      return;
+  private async updateDocumentForFile(filePath: string) {
+    const project = this.config.getConfigForFile(filePath);
+
+    if (project) {
+      const contents = await readFile(filePath, 'utf8');
+      if (contents.trim().length === 0) {
+        return undefined;
+      }
+
+      const document = parse(new Source(contents, filePath));
+      this.documentCache.set(filePath, document);
+
+      return document;
+    }
+
+    return undefined;
+  }
+
+  private getProjects() {
+    const projects = this.config.getProjects();
+
+    if (projects) {
+      return Object.values(projects);
     }
 
-    const document = parse(new Source(contents, file));
-    this.documentCache.set(file, document);
+    this.emit('error', new Error(`No projects configured in .graphqlconfig`));
+
+    return [];
+  }
+
+  private getGlobs() {
+    return this.getProjects().reduce<string[]>((glob, project) => {
+      return glob.concat(project.includes);
+    }, []);
+  }
+
+  private getSchemaPaths() {
+    return this.getProjects().reduce<string[]>((schemas, x) => {
+      if (x.schemaPath) {
+        schemas.push(x.schemaPath);
+      }
+      return schemas;
+    }, []);
+  }
+
+  private getProjectDocuments(): GraphQLProjectDocuments {
+    return Array.from(this.documentCache.entries())
+      .map(([filePath, document]) => ({
+        document,
+        filePath,
+        project: this.config.getConfigForFile(filePath),
+      }))
+      .reduce((groups, {document, filePath, project}) => {
+        if (project) {
+          const group = groups.get(project.projectName || '');
```

Comment:
`projectName` can be null if there are no projects defined in the config (i.e. single unwrapped project). empty string seemed like the safe option to default to here. We should always have a `projectName` set in practice (for our use cases). I did notice some awkwardness with `graphql-config` when using the single unwrapped project format, so I recommend against it.

Reply:
maybe we should have a validation step that errors out when when either `project` or `projectName` is falsy?

Reply:
i thought about that but i was hesitant because the `.graphqlconfig` scheme technically supports it (and they will likely fix the bugs that currently exist sometime soon)